### PR TITLE
Add CompilationEnd custom tag to diagnostics reported in compilation end actions

### PR DESCRIPTION
**Pull request by mavasani**
Required to account for IDE experience for compilation end diagnostics as a consequence of dotnet/roslyn#48762.

As many security rules are compilation end diagnostics, I have made the new parameter `isReportedAtCompilationEnd` a required parameter for `SecurityHelpers.CreateDiagnosticDescriptor` helper.

### CODE REVIEW

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/UseDefaultDllImportSearchPathsAttribute.cs [hunk 1]
```diff
@@ -23,6 +23,7 @@ public sealed class UseDefaultDllImportSearchPathsAttribute : DiagnosticAnalyzer
             RuleLevel.Disabled,
             isPortedFxCopRule: false,
             isDataflowRule: false,
+                isReportedAtCompilationEnd: false,
```
Comment (mavasani) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/UseDefaultDllImportSearchPathsAttribute.cs:26]:
```suggestion
            isReportedAtCompilationEnd: false,
```

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/UseDefaultDllImportSearchPathsAttribute.cs [hunk 2]
```diff
@@ -31,6 +32,7 @@ public sealed class UseDefaultDllImportSearchPathsAttribute : DiagnosticAnalyzer
...
```
Comment (mavasani) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/UseDefaultDllImportSearchPathsAttribute.cs:35]:
```suggestion
            isReportedAtCompilationEnd: false,
```

**Review by dotpaul — APPROVED** [review]
:ship:

**Comment by mavasani** [comment]
Thank you @dotpaul! I'll also merge the corresponding port PR soon: https://github.com/dotnet/roslyn-analyzers/pull/4360

**Pull request merged**